engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EnsoProjectNode.java

```diff
@@ -0,0 +1,135 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.Truffle;
+import com.oracle.truffle.api.TruffleFile;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import java.util.Optional;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.node.EnsoRootNode;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.callable.atom.Atom;
+import org.enso.interpreter.runtime.data.EnsoFile;
+import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.error.DataflowError;
+import org.enso.interpreter.runtime.library.dispatch.TypesLibrary;
+import org.enso.pkg.Package;
+
+@BuiltinMethod(
+    type = "Project_Description",
+    name = "enso_project_builtin",
+    description = "Returns the project description of the project given as the argument")
+public abstract class EnsoProjectNode extends Node {
+  public static EnsoProjectNode build() {
+    return EnsoProjectNodeGen.create();
+  }
+
+  /**
+   * @param module Either {@code Nothing}, or a module.
+   */
+  public abstract Object execute(Object module);
+
+  /**
+   * Fetches the second stack frame from Truffle runtime (note that the first stack frame is the
+   * call of {@code enso_project}) - the caller of {@code enso_project}, and finds in which package
+   * the caller is located.
+   *
+   * @param nothing Nothing, or interop null.
+   */
+  @Specialization(guards = "isNothing(interop, nothing)")
+  @TruffleBoundary
+  public Object getCurrentProjectDescr(
+      Object nothing, @CachedLibrary(limit = "5") InteropLibrary interop) {
+    var ctx = EnsoContext.get(this);
+    // Find the caller of `enso_project`, i.e., of this node, and find in which package
+    // it is located. The first frame is skipped, because it is always `Enso_Project.enso_project`,
+    // i.e., the first frame is always call of this specialization.
+    Optional<Package<TruffleFile>> pkgOpt =
```

Comment:
Opinion: Java code is better without `Optional`. Easier to use `null`. But I see `null` has special meaning for the iteration, so I am able to live with it.